fiftyone/utils/annotations.py

```diff
@@ -768,8 +769,26 @@ def _get_classes(
     if classes:
         return classes
 
+    return _get_existing_classes(samples, label_field)
+
+
+def _get_existing_classes(samples, label_field):
+    mapped_labels = _get_labels_map(samples, label_field)
+
     _, label_path = samples._get_label_field_path(label_field, "label")
-    return samples._dataset.distinct(label_path)
+    classes = samples._dataset.distinct(label_path)
+    mapped_classes = [mapped_labels.get(c, c) for c in classes]
```

Comment:
I would recommend a different, simpler approach here.

Just to level set, the original thinking behind using `samples._dataset` to populate a default classes list here rather than `samples` was that, if the user has constructed a view they want to edit that only contains a subset of labels, they would probably still like for the full list of classes on that label field to be made available in the annotation backend.

For example, say they have tagged some objects with incorrect labels that they want to edit, and it so happens that only `dog` labels are in the view and the desired action is to replace these with `german shepherd` or `terrier`, which are also in the full dataset schema.

The impetus for this PR is that this strategy doesn't account for the fact `samples` may be a view that dynamically modifies the labels, which I agree needs to be handled.

However, the issue with the chosen approach here is that `map_labels()` is only one of many possible ways to edit labels. For example, the view could use something like `set_field()` and `F("label").upper().concat("-OOPS")` to do the transformation. Or maybe the view just goes off the rails and uses `mongo()`...

Therefore, my recommendation would be to simply replace:

```py
classes = samples._dataset.distinct(label_path)
```

with 

```py
classes = sorted(set(samples._dataset.distinct(label_path)) | set(samples.distinct(label_path)))
```

Then we get the best of both worlds: any label in the dataset's classes OR the view's classes can be used, and there's no fancy transformations done at import time.

Reply:
Great call! This does end up making any label mapping permanent on the dataset, however, this does seem to be the best way to handle most cases. I've made these updates.